### ISSUE TITLE
Update attributes doesn't work for string IDs

### ISSUE THREAD

**Issue by pulkitsinghal** [strongloop-community/loopback-connector-elastic-search#4]
For an API URL call like: `http://localhost:3000/api/StoreConfigModels/XjEuI-pbRKWeSNFNxJRQsA` the id isn't passed through to the update attributes method and it fails:

```
{
  "error": {
    "name": "Error",
    "status": 500,
    "message": "id not set!",
    "stack": "Error: id not set!
  at [object Object].updateAttrs [as updateAttributes] (/Users/pulkitsinghal/dev/shoppinpal/loopback-connector-elasticsearch/lib/esConnector.js:887:15)
  at ModelConstructor.<anonymous> (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/dao.js:1907:27)
  at ModelConstructor.trigger (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/hooks.js:65:12)
  at ModelConstructor.<anonymous> (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/dao.js:1893:14)
  at ModelConstructor.trigger (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/hooks.js:65:12)
  at /Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/dao.js:1892:12
  at ModelConstructor.<anonymous> (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/validations.js:460:11)
  at ModelConstructor.next (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/hooks.js:75:12)
  at ModelConstructor.<anonymous> (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/validations.js:457:23)
  at ModelConstructor.trigger (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/hooks.js:65:12)
  at ModelConstructor.Validatable.isValid (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/validations.js:433:8)
  at /Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/dao.js:1886:10
  at doNotify (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/model.js:595:49)
  at doNotify (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/model.js:595:49)
  at doNotify (/Users/pulkitsinghal/dev/w2/node_modules/loopback-datasource-juggler/lib/model.js:595:49)
  ..."
  }
}
```

**Comment by pulkitsinghal** [comment]
```
  "idInjection": false,
  "properties": {
    "objectId": {
      "type": "string",
      "id": true,
      "generated": true
    },
```

**Comment by pulkitsinghal** [comment]
Added unit tests for string IDs and they work:
https://github.com/strongloop-community/loopback-connector-elastic-search/commit/5fc90e2e7bcfcb7053b10154f33ef010038a2e9d

So then the problem I have when I try to use this module, must be on the angular side? Is there a known bug around this in angular @raymondfeng?

**Comment by pulkitsinghal** [comment]
I could expose the feature branch where I try to use the es connector but there are a few configuration files that need to be setup, if anyone wants to leverage my existing project to see this problem in action, they would have to coordinate with me over gitter a bit...

**Comment by pulkitsinghal** [comment]
@raymondfeng - any quick comments on this one?

**Comment by pulkitsinghal** [comment]
I believe this may have been inadvertently fixed alongside issue #3
